packages/edit-site/src/components/left-sidebar/navigation-panel/template-navigation-items.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import TemplatePreview from './template-preview';
+import { NavigationPanelPreviewFill } from '.';
+import { ITEM_CONTENTS } from './constants';
+
+export default function TemplateNavigationItems( {
+	entityType = 'wp_template',
+	templates,
+	onActivate,
+} ) {
+	const [ hoveredTemplate, setHoveredTemplate ] = useState();
+
+	const onMouseEnterTemplate = ( template ) => setHoveredTemplate( template );
+	const onMouseLeaveTemplate = () => setHoveredTemplate( null );
+
+	if ( ! templates ) {
+		return null;
+	}
+
+	if ( ! Array.isArray( templates ) ) {
+		templates = [ templates ];
+	}
+
+	const itemPrefix =
+		entityType === 'wp_template_part' ? 'template-part' : 'template';
```

Comment:
Just realized that we don't really need this "conversion". 🤔 
We could just simply use the template type + the ID.

E.g.
```jsx
item={ `${ entityType }-${ template.id }` }
```

(Here, and in the NavigationPanel as well)